File: sydney/Kernel/Common/ErrorLevel.cpp
// -*-Mode: C++; tab-width: 4; c-basic-offset: 4;-*-
// vi:set ts=4 sw=4:
//
// ErrorLevel.cpp -- エラーレベルをあらわすクラス
// 
// Copyright (c) 2005, 2023 Ricoh Company, Ltd.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// 

#include "SyDefault.h"
#include "SyReinterpretCast.h"

#include "Common/ClassID.h"
#include "Common/ErrorLevel.h"

_TRMEISTER_USING
_TRMEISTER_COMMON_USING

//
//	FUNCTION public
//	Common::ErrorLevel::ErrorLevel -- コンストラクタ(1)
//
//	NOTES
//
//	ARGUMENTS
//	なし
//
//	RETURN
//	なし
//
//	EXCEPTIONS
//	なし
//
ErrorLevel::ErrorLevel()
	: m_eLevel(Undefined)
{
}

//
//	FUNCTION public
//	Common::ErrorLevel::ErrorLevel -- コンストラクタ(2)
//
//	NOTES
//
//	ARGUMENTS
//	Common::ErrorLevel::Type eLevel_
//		エラーレベル
//
//	RETURN
//	なし
//
//	EXCEPTIONS
//	なし
//
ErrorLevel::ErrorLevel(Type eLevel_)
	: m_eLevel(eLevel_)
{
}

//
//	FUNCTION public
//	Common::ErrorLevel::~ErrorLevel -- デストラクタ
//
//	NOTES
//
//	ARGUMENTS
//	なし
//
//	RETURN
//	なし
//
//	EXCEPTIONS
//	なし
//
ErrorLevel::~ErrorLevel()
{
}

//
//	FUNCTION public
//	Common::ErrorLevel::getErrorLevel -- エラーレベルを得る
//
//	NOTES
//
//	ARGUMENTS
//	なし
//
//	RETURN
//	Common::ErrorLevel::Type
//	   エラーレベル
//
//	EXCEPTIONS
//	なし
//
ErrorLevel::Type
ErrorLevel::getLevel() const
{
	return m_eLevel;
}

//
//	FUNCTION public
//	Common::ErrorLevel::serialize -- シリアル化のためのメソッド
//
//	NOTES
//	シリアル化を行う
//
//	ARGUMENTS
//	ModSrchive& cArchiver_
//		アーカイバ
//
//	RETURN
//	なし
//
//	EXCEPTIONS
//	その他
//		下位の例外はそのまま再送
//
void
ErrorLevel::serialize(ModArchive& cArchiver_)
{
	if (cArchiver_.isStore())
	{
		//書出し
		int iLevel = m_eLevel;
		cArchiver_ << iLevel;
	}
	else
	{
		//読出し
		int iLevel;
		cArchiver_ >> iLevel;
		m_eLevel = static_cast<Type>(iLevel);
	}
}

//
//	FUNCTION public
//	Common::ErrorLevel::getClassID -- クラスIDを得る
//
//	NOTES
//	クラスIDを得る
//
//	ARGUMENTS
//	なし
//
//	RETURN
//	int
//		クラスID
//
//	EXCEPTIONS
//	なし
//
int
ErrorLevel::getClassID() const
{
	return ClassID::ErrorLevelClass;
}

//
//	Copyright (c) 2005, 2023 Ricoh Company, Ltd.
//	All rights reserved.
//

File: sydney/Java/JDBC/src/jp/co/ricoh/doquedb/common/ErrorLevel.java
// -*-Mode: Java; tab-width: 4; c-basic-offset: 4;-*-
// vi:set ts=4 sw=4:
//
// ErrorLevel.java -- エラーレベルをあらわすクラス
//
// Copyright (c) 2005, 2023 Ricoh Company, Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

package jp.co.ricoh.doquedb.common;

/**
 * エラーレベルをあらわすクラス
 *
 */
public final class ErrorLevel
	implements Serializable
{
	/** ユーザレベル */
	public final static int USER = 1;
	/** システムレベル */
	public final static int SYSTEM = 2;
	/** 不明なエラーレベル値 */
	public final static int UNDEFINED = -1;

	/** 値 */
	private int _level;

	/**
	 * 新たにエラーレベルを作成する。
	 */
	public ErrorLevel()
	{
		_level = UNDEFINED;
	}

	/**
	 * 新たにエラーレベルを作成する。
	 *
	 * @param level_	格納する値
	 */
	public ErrorLevel(int level_)
	{
		_level = level_;
	}

	/**
	 * エラーレベルを得る
	 *
	 * @return	格納されているエラーレベル
	 */
	public int getLevel()
	{
		return _level;
	}

	/**
	 * エラーレベルを設定する
	 *
	 * @param level_	格納するエラーレベル
	 */
	public void setLevel(int level_)
	{
		_level = level_;
	}

	/**
	 * ユーザレベルかどうか
	 *
	 * @return	ユーザレベルの場合はtrue、それ以外の場合はfalse
	 */
	public boolean isUserLevel()
	{
		return (_level == USER);
	}

	/**
	 * ストリームから読み込む
	 *
	 * @param input_	入力用のストリーム
	 * @throws java.io.IOException
	 *			入出力関係の例外が発生した
	 * @see Serializable#readObject(InputStream) readObject
	 */
	public void readObject(InputStream input_)
		throws java.io.IOException
	{
		_level = input_.readInt();
	}

	/**
	 * ストリームに書き出す
	 *
	 * @param output_	出力用のストリーム
	 * @throws java.io.IOException
	 *			入出力関係の例外が発生した
	 * @see	Serializable#writeObject(OutputStream) writeObject
	 */
	public void writeObject(OutputStream output_)
		throws java.io.IOException
	{
		output_.writeInt(_level);
	}

	/**
	 * {@link ClassID クラスID}を得る
	 *
	 * @return {@link ClassID クラスID}
	 * @see Serializable#getClassID() getClassID
	 */
	public int getClassID()
	{
		return ClassID.ERROR_LEVEL;
	}
}

//
// Copyright (c) 2005, 2023 Ricoh Company, Ltd.
// All rights reserved.
//
